src/NBitcoin/IBitcoinSerializable.cs

```diff
@@ -9,34 +9,44 @@ public interface IBitcoinSerializable
         void ReadWrite(BitcoinStream stream);
     }
 
-    public static class BitcoinSerializableExtensions
+    public interface IHaveNetworkOptions
     {
-        public static void ReadWrite(this IBitcoinSerializable serializable, Stream stream, bool serializing, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION)
-        {
-            BitcoinStream s = new BitcoinStream(stream, serializing)
+        NetworkOptions GetNetworkOptions();
+    }
+
+	public static class BitcoinSerializableExtensions
+	{
+		public static void ReadWrite(this IBitcoinSerializable serializable, Stream stream, bool serializing, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION, 
+            NetworkOptions options = null)
+		{
+            // If no options have been provided then take the options from the serializable
+            if (options == null && serializing && serializable is IHaveNetworkOptions)
+                options = (serializable as IHaveNetworkOptions).GetNetworkOptions();
+
+            serializable.ReadWrite(new BitcoinStream(stream, serializing)
```

Comment:
I think there may be many places where the options may not be readily available to pass to the BitcoinStream constructor. Maybe in those scenarios serialization is not dependent on options either. I've added a TODO so that we can revisit this.